records_mover/records/schema/field/__init__.py

```diff
@@ -188,8 +188,10 @@ def cast_series_type(self, series: 'Series') -> 'Series':
                     # https://stackoverflow.com/questions/34501930/how-to-convert-timedelta-to-time-of-day-in-pandas
 
                     # Convert from "0 days 12:34:56.000000000" to "12:34:56"
-                    def components_to_time_str(df: pd.DataFrame) -> str:
-                        return f"{df['hours']:02d}:{df['minutes']:02d}:{df['seconds']:02d}"
+                    def components_to_time_str(df: pd.DataFrame) -> datetime.time:
+                        return datetime.time(hour=df['hours'],
+                                             minute=df['minutes'],
+                                             second=df['seconds'])
```

Comment:
After  longer restricting the timeonlyformat, the MySQL integration test failed.

MySQL exports its time columns as timedelta objects (they cleverly decided their TIME type means either 'time of day' or 'difference between two times'.  Thanks, MySQL!

We previously preformatted those as strings when we found them - but a better plan now that we can format datetime.time() objects is to send it along as that and let prep_for_csv.py handle them.

This was reproduced by the integration tests:

https://app.circleci.com/pipelines/github/bluelabsio/records-mover/1823/workflows/ef8242cd-55c3-4442-b72b-37bea71c89eb/jobs/15583

I then wrote a component test, which failed like this:

https://app.circleci.com/pipelines/github/bluelabsio/records-mover/1855/workflows/dc6a566c-413f-43a0-ae1c-6b2b114f0b40/jobs/15894